### PR TITLE
Integration tests on spark 3.0.1-SNAPSHOT & 3.1.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by NvTimLiu**
1, Add integration tests Jenkinsfile on spark 3.0.1-SNAPSHOT & 3.1.0-SNAPSHOT

2, Only update Jenkins scripts, no source change. No unit tests needed.

3, No Github issue related to the PR.

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

File: jenkins/Jenkinsfile.310.integration [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+*
+* Jenkins file for running spark3.0 integration tests
```
Comment (jlowe) [inline jenkins/Jenkinsfile.310.integration:20]:
```suggestion
* Jenkins file for running spark3.1 integration tests
```

File: jenkins/Jenkinsfile.301.integration [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+*
+* Jenkins file for running spark3.0 integration tests
```
Comment (jlowe) [inline jenkins/Jenkinsfile.301.integration:20]:
```suggestion
* Jenkins file for running spark3.0.1 integration tests
```

File: jenkins/Jenkinsfile.310.integration [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        LIBCUDF_KERNEL_CACHE_PATH='/tmp/.cudf'
+        ARTIFACTORY_NAME = "${ArtifactoryConstants.ARTIFACTORY_NAME}"
+        URM_URL = "${urmUrl}"
+        MVN_URM_MIRROR='-s jenkins/settings.xml -P mirror-apache-to-urm'
+    }
+
+    stages {
+        stage('IT on 3.1.0-SNAPSHOT') {
+            agent { label 'docker-gpu' }
+            environment {SPARK_VER='3.0.1-SNAPSHOT'}
```
Comment (jlowe) [inline jenkins/Jenkinsfile.310.integration:56]:
```suggestion
            environment {SPARK_VER='3.1.0-SNAPSHOT'}
```

File: jenkins/Jenkinsfile.310.integration [hunk 4]
```diff
@@ -0,0 +1,99 @@
+#!/usr/local/env groovy
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
```
Comment (jlowe) [inline jenkins/Jenkinsfile.310.integration:3]: 2019 copyrights?

**Comment by NvTimLiu** [comment]
> There's a lot of duplication across Jenkins files. Not must-fix for this PR, but it would be nice to have a followup to refactor this if possible. Otherwise every time we want to tweak something in the integration test pipelines we need to remember to make that same change in a bunch of other files (e.g.: updating the `slack` method).

Good suggestion!
I'll investigate how to refactor these Jenkinsfiles

File: jenkins/Jenkinsfile.301.integration [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#!/usr/local/env groovy
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
```
[the remark this answers is not in the document]
Comment (NvTimLiu) [reply jenkins/Jenkinsfile.301.integration]: Fixed

File: jenkins/Jenkinsfile.310.integration (hunk 1 above)
Comment (NvTimLiu) [reply jenkins/Jenkinsfile.310.integration]: fixed

File: jenkins/Jenkinsfile.301.integration (hunk 2 above)
Comment (NvTimLiu) [reply jenkins/Jenkinsfile.301.integration]: fixed

File: jenkins/Jenkinsfile.310.integration (hunk 3 above)
Comment (NvTimLiu) [reply jenkins/Jenkinsfile.310.integration]: fixed

File: jenkins/Jenkinsfile.310.integration (hunk 4 above)
Comment (NvTimLiu) [reply jenkins/Jenkinsfile.310.integration]: fixed

**Comment by NvTimLiu** [comment]
build

**Comment by revans2** [comment]
build

**Pull request merged**